tests/Doctrine/Tests/DBAL/Functional/StatementTest.php

```diff
@@ -320,4 +323,65 @@ public function testFetchInColumnMode(): void
 
         self::assertEquals(1, $result);
     }
+
+    public function testFetchAllWithOneArgumentOnExternalPDO(): Connection
+    {
+        $driver = $this->connection->getDriver();
+
+        if (! $driver instanceof PDOSqliteDriver) {
+            $this->markTestSkipped('External PDO connection tests run only with an SQLite configuration.');
+        }
+
+        $pdo = new PDO('sqlite::memory:');
+
+        $connection = new Connection(['pdo' => $pdo], new PDOSqliteDriver());
+
+        $table = new Table('stmt_fetch_all');
+        $table->addColumn('a', 'integer');
+        $table->addColumn('b', 'integer');
+        $connection->getSchemaManager()->createTable($table);
+
+        $connection->insert('stmt_fetch_all', [
+            'a' => 1,
+            'b' => 2,
+        ]);
+
+        $stmt = $connection->prepare('SELECT a, b FROM stmt_fetch_all');
+        $stmt->execute();
+
+        self::assertEquals([[1, 2]], $stmt->fetchAll(FetchMode::NUMERIC));
+
+        return $connection;
+    }
+
+    /**
+     * @depends testFetchAllWithOneArgumentOnExternalPDO
```

Comment:
Why does the test for one mode depend on the test for another? If it relies on the setup, then it should be extracted and shared. Ideally, it should be a separate throwaway test with a proper setup implementation.

Reply:
I've moved the tests to their own class, with proper `setUp()`.